### ISSUE TITLE
Add support for GitLab.

### ISSUE THREAD

**Issue by saschagrunert** [mversluys/premake-ghp#3]
Hey 👋🏻,

How about general URL or GitLab support? Also specifying the dependencies in a TOML file would be a great addition. 

What do you think?

**Comment by mversluys** [comment]
The format of references would need to be extended to be more like go package references, which would be straight forward. Does GitLab make it easy to retrieve the repository in a zip file? The GHP index is making use of GitHub APIs at the moment, if there are similar for GitLab then that could be added. Pull requests for such changes would certainly be looked at 😊

**Comment by mversluys** [comment]
Regarding TOML, the intent was to declare directly in the premake lua file which means there's nothing new for people to learn, and it's bundled directly with the other premake directives.

**Comment by saschagrunert** [comment]
Yeah, I agree with the toml thing. Yes, GitLab allows repository downloads via the API or [the download button here](https://gitlab.com/gitlab-org/gitlab-ce/tree/master). I am not ready to provide a pull request yet. What I implemented is, that my premake extension provides a function `add_dependency(url)`, which assumes a tarball including something like: 

```
./include/*.h
./lib/*.a
```

These files will be copied to a local `vendor` directory with the same structure. Then premake links agains these libraries.

Do you think that this is a good addition to your project?

**Comment by saschagrunert** [comment]
Also a go like Project Download and Compilation Option would also be cool. But to minimize the compile time such an precompiled alternative could be reasonable.
